go.mod

```diff
@@ -17,21 +20,21 @@ require (
 	github.com/mitchellh/go-ps v1.0.0
 	github.com/muvaf/typewriter v0.0.0-20210910160850-80e49fe1eb32
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus/client_golang v1.14.0
+	github.com/prometheus/client_golang v1.15.1
 	github.com/spf13/afero v1.9.2
 	github.com/tmccombs/hcl2json v0.3.3
 	github.com/yuin/goldmark v1.4.13
 	github.com/zclconf/go-cty v1.11.0
-	golang.org/x/net v0.7.0
+	golang.org/x/net v0.10.0
 	gopkg.in/alecthomas/kingpin.v2 v2.2.6
 	gopkg.in/yaml.v2 v2.4.0
 	gopkg.in/yaml.v3 v3.0.1
-	k8s.io/api v0.26.3
-	k8s.io/apimachinery v0.26.3
-	k8s.io/client-go v0.26.3
-	k8s.io/utils v0.0.0-20221128185143-99ec85e7a448
-	sigs.k8s.io/controller-runtime v0.14.6
-	sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2
+	k8s.io/api v0.27.2
+	k8s.io/apimachinery v0.27.2
+	k8s.io/client-go v0.27.2
+	k8s.io/utils v0.0.0-20230505201702-9f6742963106
+	sigs.k8s.io/controller-runtime v0.15.0
+	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd
```

Comment:
Are all of these dependency updates needed for the PR? 
I only ask as I saw the Renovate PR to bump these caused our pipelines to all fail https://github.com/upbound/provider-aws/pull/759

Reply:
it is because the controller-runtime was updated in xp-runtime and that pulls all of the other stuff. Ill make a separate PR for that to make this PR leaner

Reply:
actually, ill leave it for now. When https://github.com/crossplane/crossplane-runtime/pull/456 gets merged, we can do a xp-runtime release, so that here we get the dependency on a released version, instead of a commit

Reply:
> Are all of these dependency updates needed for the PR? I only ask as I saw the Renovate PR to bump these caused our pipelines to all fail [upbound/provider-aws#759](https://github.com/upbound/provider-aws/pull/759)

Now i checked, when testing i had to update one line [here](https://github.com/upbound/provider-aws/blob/1ef9fd0c17313478f30b9bcd3d48701e0b1ec9f7/internal/controller/providerconfig/config.go#L33) for `provider-aws` . Probably its failing because of that:

```
-               Watches(&source.Kind{Type: &v1beta1.ProviderConfigUsage{}}, &resource.EnqueueRequestForProviderConfig{}).
+               Watches(&v1beta1.ProviderConfigUsage{}, &resource.EnqueueRequestForProviderConfig{}).
```